packages/@react-spectrum/s2/src/AvatarGroup.tsx

```diff
@@ -0,0 +1,163 @@
+
+  /** The label for the avatar group. */
+  label?: string,
+
+  /**
+   * The size of the avatar group.
+   *
+   * @default 'M'
+   * */
+  size?: 'S' | 'M' | 'L' | 'XL' | 'XXL' | 'XXXL',
+  /* stroke color (should match the container background color) */
+
+  strokeColor?: 'base' | 'layer-1' | 'layer-2' | 'pasteboard'
+}
+
+interface AvatarGroupItemProps {
+  /** Text description of the avatar. */
+  alt?: string,
+  /** The image URL for the avatar. */
+  src: string
+}
+
+interface AvatarGroupContextProps extends Pick<AvatarGroupProps, 'strokeColor'> {
+  styles?: StyleString
+}
+
+const avatar = style({
+  marginStart: {
+    default: {
+      size: {
+        S: size(-5),
+        M: size(-6),
+        L: size(-7),
+        XL: size(-8),
+        XXL: size(-9),
+        XXXL: size(-10)
+      }
+    },
+    isFirst: 0
+  },
+  size: {
+    size: {
+      S: 20,
+      M: 24,
+      L: 28,
+      XL: 32,
+      XXL: 36,
+      XXXL: 40
+    }
+  }
+});
+
+const text = style({
+  marginStart: 8,
+  fontFamily: 'sans',
+  fontSize: {
+    size: {
+      S: 'ui-sm',
+      M: 'ui',
+      L: 'ui-lg',
+      XL: 'ui-xl',
+      XXL: 'ui-2xl',
+      XXXL: 'ui-3xl'
+    }
+  }
+});
+
+const imageStyles = style({
+  borderRadius: 'full',
+  borderWidth: 1,
+  borderStyle: 'solid',
+  '--avatarBorder': {
+    type: 'backgroundColor',
+    value: {
+      strokeColor: {
+        base: 'base',
+        'layer-1': 'layer-1',
+        'layer-2': 'layer-2',
+        pasteboard: 'pasteboard'
+      }
+    }
+  },
+  borderColor: '[var(--avatarBorder)]',
+  size: 20,
+  disableTapHighlight: true
+}, getAllowedOverrides({height: true}));
+
+const AvatarGroupContext = createContext<AvatarGroupContextProps>({});
+
+const container = style({
+  display: 'flex',
+  alignItems: 'center',
+  listStyleType: 'none',
+  paddingStart: 0,
+  margin: 0
+}, getAllowedOverrides());
+
+function AvatarGroup(props: AvatarGroupProps) {
+  let {children, label, size = 'M', strokeColor = 'base', styles, UNSAFE_style, UNSAFE_className, ...otherProps} = props;
+  let labelId = useId();
+
+  return (
+    <ul
+      aria-labelledby={labelId}
+      {...filterDOMProps(otherProps)}
+      tabIndex={0}
+      role="group"
+      className={(UNSAFE_className ?? '') + container(null, styles)}
+      style={UNSAFE_style}>
+      {
+        Children.map(children, (child, index) => (
+          <AvatarGroupContext.Provider value={{styles: avatar({size, isFirst: index === 0}), strokeColor}}>
+            {child}
+          </AvatarGroupContext.Provider>
+        ))
+      }
+      {label && <span id={labelId} className={text({size})}>{label}</span>}
+    </ul>
+  );
+}
+
+let _AvatarGroup = forwardRef(AvatarGroup);
+export {_AvatarGroup as AvatarGroup};
+
+export function AvatarGroupItem(props: AvatarGroupItemProps) {
+  let {
+    alt = '',
+    src
+  } = props;
+  let {styles, strokeColor} = useContext(AvatarGroupContext);
+  return (
+    <li className={style({display: 'flex'})}>
```

Comment:
why display flex?

Reply:
without flex, the list item element looks like this:
![Screenshot 2024-08-09 at 2 10 05 PM](https://github.com/user-attachments/assets/5dccb32e-c20f-4865-92cb-4da0a2619b69)

definitely felt weird using it to fix the styling, so definitely open to another solution

Reply:
i had an issue with the dimensions of the avatars in the `<li>` element, but since i've removed it, it works fine without the flex